### PR TITLE
Workaround for excessively long gcc command line build issue

### PR DESCRIPTION
**Pull request by dmitri-d**
This is a port of scripts and patches used during proxy rpm package build

### CODE REVIEW

**Comment by dmitri-d** [comment]
> Why this can't be checked-in in tree? So that an user building it locally could benefit?

Upstream didn't update include paths until a later envoy version. Should we modify paths in the codebase directly, none of upstream changes (bug fixes and CVE fixes) would apply without changes.

> You did it on envoy side already, no?

Not yet.

File: maistra/ci/post-submit.sh [hunk 1]
```diff
@@ -12,6 +12,9 @@ DIR=$(cd "$(dirname "$0")" ; pwd -P)
 # shellcheck disable=SC1090
 source "${DIR}/common.sh"
 
+# modify include paths to avoid overly long gcc command line
+fix_include_paths()
```
Comment (jwendell) [inline maistra/ci/post-submit.sh]: you can move this call (without `()`) to the common.sh file

File: maistra/ci/pre-submit.sh [hunk 2]
```diff
@@ -12,6 +12,9 @@ DIR=$(cd "$(dirname "$0")" ; pwd -P)
 # shellcheck disable=SC1090
 source "${DIR}/common.sh"
 
+# modify include paths to avoid overly long gcc command line
+fix_include_paths
```
Comment (jwendell) [inline maistra/ci/pre-submit.sh]: you can move this call to the common.sh file

**Review by jwendell — COMMENTED** [review]
ah ok.
so in https://github.com/maistra/envoy/pull/98 we added the envoy files but are not using them? should it be reverted in favor of this one then?

**Comment by dmitri-d** [comment]
> so in maistra/envoy#98 we added the envoy files but are not using them? should it be reverted in favor of this one then?

Lol, I completely forgot about it, updated the PR to use the patch and the script in maistra-envoy repository.

File: maistra/ci/post-submit.sh (hunk 1 above)
Comment (dmitri-d) [reply maistra/ci/post-submit.sh]: This and below: not sure what you mean -- paths should only be modified once per checkout, can't invoke the update from bazel_build/bazel_test...

**Comment by dmitri-d** [comment]
@jwendell: the build is going to fail, there's a syntax error in the script used to update paths in envoy (the fix is here: https://github.com/maistra/envoy/pull/106)

File: maistra/ci/post-submit.sh (hunk 1 above)
Comment (jwendell) [reply maistra/ci/post-submit.sh]: You are calling this function in both files. You can just move this call to the bottom of common.sh, which is included in pre and post submit files.

Comment (dmitri-d) [reply maistra/ci/post-submit.sh]: done

**Review by jwendell — APPROVED** [review]

**Pull request merged**